### PR TITLE
Fix access token issue on some public endpoints

### LINKED ISSUE #24159: /api/v1/orgs has become an authenticated endpoint with 0.19
### Description

We have CI for https://opendev.org where we are testing our upgrade of 1.18.5 to 1.19.1 [1].

We found that one of our CI tasks that gets the orgs from ```api/v1/orgs``` started failing returning a ```401```.

I believe this is related to https://github.com/go-gitea/gitea/commit/de484e86bc495a67d2f122ed438178d587a92526

It's not too hard to replicate this; as opendev.org is still on 1.18

```
$ curl https://opendev.org/api/v1/orgs
$ curl https://try.gitea.io/api/v1/orgs
```

The first returns json, the second not :)  I'm not sure if this is intentional; it does seem like the public org listing probably doens't need to be an authenticated call?

[1] https://review.opendev.org/c/opendev/system-config/+/877541

### Gitea Version

1.19.1

### Can you reproduce the bug on the Gitea demo site?

Yes

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

Our gitea is built from upstream into a container

### Database

None

### PR DESCRIPTION
**Pull request by harryzcy**
- [x] Identify endpoints that should be public
- [x] Update integration tests

Fix #24159

### CODE REVIEW

File: tests/integration/api_org_test.go [hunk 1]
```diff
@@ -144,9 +144,7 @@ func TestAPIOrgDeny(t *testing.T) {
 func TestAPIGetAll(t *testing.T) {
 	defer tests.PrepareTestEnv(t)()
 
-	token := getUserToken(t, "user1", auth_model.AccessTokenScopeReadOrg)
-
-	req := NewRequestf(t, "GET", "/api/v1/orgs?token=%s", token)
+	req := NewRequestf(t, "GET", "/api/v1/orgs") // token not required
```
Comment (delvh) [inline tests/integration/api_org_test.go]: It is required for limited and private repos.
Perhaps we should add a test case for that.

Comment (harryzcy) [reply tests/integration/api_org_test.go]: Do you mean it should return public repo when accessing without a token, and return all repos when using a token with correct scope?
That will complicate things a lot, and many fixes are needed.

Comment (delvh) [reply tests/integration/api_org_test.go]: No, I simply had a look into `GetAll`.
The logic is
`return public repos only, if user is signed in limited as well, and if they are even an admin, they get everything`.
That's what I meant.
At the moment, we simply test for the public repos.

**Review by lunny — APPROVED** [review]

**Review by wolfogre — APPROVED** [review]

**Comment by techknowlogick** [comment]
🤖 🎺

**Pull request merged**